terraform/files/bin/create_cluster.sh

```diff
@@ -149,15 +168,12 @@ fi
 echo "# apply configuration and deploy cluster ${CLUSTER_NAME}"
 kubectl apply -f ~/${CLUSTER_NAME}/${CLUSTER_NAME}-config.yaml || exit 3
 
-#Waiting for Clusterstate Ready
+# Waiting for Cluster=Ready
+wait_for_k8s_resource_matching kubeadmcontrolplanes/${CLUSTER_NAME}-control-plane
+wait_for_k8s_resources_matching machines/${CLUSTER_NAME} "machines -l cluster.x-k8s.io/cluster-name=${CLUSTER_NAME}"
+
 echo "# Waiting for Cluster=Ready"
```

Comment:
Suggestion: Move this *cluster* **echo**(line 175) three lines above as well as *kube-proxy* **echo**(line 199).

Reply:
I put the wait_* calls above the echos because while waiting for e.g. `kubeadmcontrolplanes/${CLUSTER_NAME}-control-plane` it is printing `[0] waiting for kubeadmcontrolplanes/${CLUSTER_NAME}-control-plane` while the below `kubectl wait` executions prints nothing and have a timeout of e.g. 5 minutes where you are left in the dark what is happening. For debugging purposes you know this way that it already waited successfully for the resources to appear (wait_* calls) but is now at the `kubectl wait` stage.

That being said let me know if you still prefer it the other way around. I can move the lines up easily :smile:

Reply:
Thank you, it makes sense to me. We can keep it as is.